### ISSUE TITLE
How to export In PHPExcel without rendering datagrid before

### ISSUE THREAD

**Issue by jonathangreco** [ThaDafinser/ZfcDatagrid#180]
I have a custom SQL request, give me an array, It's my datasource. From this result without showing to the user the table, i need to directly create the PHPExcel file (wich been attached to a mail and send in the same script, this part is done)

I did some work, it works twice, but i have this problem 

`Sort conditions from cache are missing!`

This is my code of my function :
``````PHP
        $grid->setDataSource($results);
        $this->designDataGrid($grid); // Method wich add All my Columns

        $grid->setTitle('some_random_title');
        $grid->setRendererName('PHPExcel');
        $grid->loadData();
        $grid->render();

        return $grid->getCacheId() . '.xlsx';
``````

EDIT : This works when i do this :

I put `loadData` before set `RendererName`, this create cache file, then i comment this one and put `loadData` after. And the file is created, but for now i don't know how to make this works 'normally'. Any help appreciated.

**Comment by ThaDafinser** [comment]
@jonathangreco it was designed to be a user action. Check pls this new PR: #197

**Comment by ThaDafinser** [comment]
Solved with #197
